### DISCUSSION TITLE
Why does rlVertex2f() not use z=0?

### DISCUSSION THREAD

**Discussion by paulmelis** [raysan5/raylib#4162] [Q&A]
It seems that for `GRAPHICS_API_OPENGL_33` the function `rlglvertex2f()` actually [calls](https://github.com/raysan5/raylib/blob/5ede47618bd9f9a440af648da1b4817e51644994/src/rlgl.h#L1530)

```
rlVertex3f(x, y, RLGL.currentBatch->currentDepth);
```

Yet for `GRAPHICS_API_OPENGL_11` it [calls](https://github.com/raysan5/raylib/blob/5ede47618bd9f9a440af648da1b4817e51644994/src/rlgl.h#L1411)

```
glVertex2f(x, y);    // z is implicitly 0
```

Why is z not consistently 0? And any way to control the `currentDepth` value? 

Edit: some background is that I'd like to use some of the 2D functions, such as `DrawSplineBasis()` in 3D, by transforming the XY coordinates they produce to whatever I need. Yet, they call `rlVertex2f()`, so the Z value seems to be out of my control

**Comment by Hex4dec** [comment]
Well, this is C, no? Just set the RLGL.currentBatch->currentDepth variable yourself! We don't have any private variables!

**Comment by paulmelis** [comment]
It's there for a reason, and I'd like to understand why. 

Also, `RLGL` is not available from rlgl.h unless you define `RLGL_IMPLEMENTATION` before including it. But rcore.c already does that, so the variable is effectively hidden (as it's also marked `static`)
